tests/introspection/tests.py

```diff
@@ -53,6 +53,14 @@ def test_table_names_with_views(self):
             with connection.cursor() as cursor:
                 cursor.execute('DROP VIEW introspection_article_view')
 
+    def test_only_nonempty_tables(self):
+        tables = connection.introspection.django_table_names()
+        nonempty = connection.introspection.only_nonempty_tables(tables)
+        self.assertEqual(nonempty, [])
+        City.objects.create(name="London")
+        nonempty = connection.introspection.only_nonempty_tables(tables)
+        self.assertEqual(nonempty, ['introspection_city'])
```

Comment:
I'd split this into two tests s o they can fail independently: `test_only_nonempty_tables_empty` and `test_only_nonempty_tables_one` or similarly named